client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -103,24 +104,146 @@ export const siteSetupFlow: Flow = {
+			}
+			case SiteIntent.Sell: {
+				switch ( currentStep ) {
+					case 'options':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 6 } } );
+						break;
+					case 'storeFeatures':
+						updateFlowProgress( { middleProgress: { progress: 2, count: 6 } } );
+						break;
+				}
+				if ( storeType === 'simple' ) {
+					switch ( currentStep ) {
+						case 'designSetup':
+							updateFlowProgress( { middleProgress: { progress: 3, count: 6 } } );
+							break;
+						case 'processing':
+							updateFlowProgress( { endStep: currentStep } );
+							break;
+					}
+				} else if ( storeType === 'power' ) {
+					switch ( currentStep ) {
+						case 'storeAddress':
+							updateFlowProgress( { middleProgress: { progress: 3, count: 6 } } );
+							break;
+						case 'businessInfo':
+							updateFlowProgress( { middleProgress: { progress: 4, count: 6 } } );
+							break;
+						case 'wooConfirm':
+						case 'wooInstallPlugins':
+							updateFlowProgress( { middleProgress: { progress: 5, count: 6 } } );
+							break;
+						case 'processing':
+							updateFlowProgress( { endStep: currentStep } );
+							break;
+					}
+				}
+
+				break;
+			}
+			case SiteIntent.Import: {
+				switch ( currentStep ) {
+					case 'import':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 4 } } );
+						break;
+					case 'importList':
+					case 'importReady':
+					case 'importReadyNot':
+					case 'importReadyWpcom':
+					case 'importReadyPreview': {
+						updateFlowProgress( { middleProgress: { progress: 2, count: 4 } } );
+						break;
+					}
+					case 'importerWix':
+					case 'importerBlogger':
+					case 'importerMedium':
+					case 'importerSquarespace':
+					case 'importerWordpress':
+						updateFlowProgress( { middleProgress: { progress: 3, count: 4 } } );
+						break;
+					case 'processing':
+						updateFlowProgress( { endStep: currentStep } );
+						break;
+				}
+
+				break;
 			}
-		} else {
-			setStepProgress( undefined );
+		}
+
+		function updateFlowProgress( flowProgress: {
+			beginningStep?: string;
+			middleProgress?: { progress: number; count: number };
+			endStep?: string;
+		} ) {
+			const beginningSteps = [
+				...( goalsStepEnabled ? [ 'goals' ] : [] ),
+				...( verticalsStepEnabled ? [ 'vertical' ] : [] ),
+				...( ! goalsStepEnabled ? [ 'intent' ] : [] ),
+			];
+
+			let beginningSegment = ( beginningSteps.length - 1 ) / MAX_STEPS;
+			let middleSegment = 0;
+
+			if ( flowProgress.beginningStep ) {
+				beginningSegment = beginningSteps.indexOf( flowProgress.beginningStep );
+			}
+
+			if ( ! flowProgress.beginningStep && flowProgress.middleProgress ) {
+				middleSegment =
+					( flowProgress.middleProgress.progress / flowProgress.middleProgress.count ) * MAX_STEPS;
+			}
+
+			let progress = beginningSegment + middleSegment;
+
+			if ( flowProgress.endStep ) {
+				progress = MAX_STEPS;
+			}
+
+			setStepProgress( { progress: progress, count: MAX_STEPS } );
```

Comment:
If the count is always `MAX_STEPS`, I think we don't need to set it all the time and we might only need to set the progress here?

Reply:
I prefer to keep this count value because MAX_STEPS is used for this use case only, there will be other cases that the count value is dynamic, what do you think?